tensorflow/python/keras/metrics.py

```diff
@@ -2272,9 +2272,14 @@ def update_state(self, y_true, y_pred, sample_weight=None):
     Returns:
       Update op.
     """
-    # Converted y_true and y_pred into Tensors
-    y_true = ops.convert_to_tensor(y_true)
-    y_pred = ops.convert_to_tensor(y_pred)
+    
+    # If y_true and y_pred are lists then 
+    # they are converted into Tensors 
+    if type(y_true) == list:
+      y_true = ops.convert_to_tensor(y_true)
+      
+    if type(y_pred) == list:
+      y_pred = ops.convert_to_tensor(y_pred)
```

Comment:
I am sorry to interrupt, but do they look good? @pavithrasv

Reply:
Thank you @gshashank84. May be what we need is: 
    y_true = math_ops.cast(y_true, self._dtype)
    y_pred = math_ops.cast(y_pred, self._dtype)
Please take a look at the other metric classes `update_state` function.